### ISSUE TITLE
Better documentation on `num_workers`, `worker_work_ratio`

### ISSUE THREAD

**Issue by ben-pr-p** [beatrichartz/parallel_stream#8]
Hi!

Thanks for the library!

Could you write up some documentation on how those options work for `ParallelStream.map`?

My intuition says that as long as `num_workers: 1`, it's effectively not parallel, and `worker_work_ratio` is for managing the batch size passed to each worker for fine tuning acceptable parallel overhead?

**Comment by beatrichartz** [comment]
Yes you're right with your assumptions as well as the assertion that these could be documented better.

**Comment by Jdcorley** [comment]
Could I take this as a first-time open source contribution?

**Comment by beatrichartz** [comment]
@Jdcorley absolutely, have at it!

**Comment by phewitt** [comment]
Did this ever happen? I think another bit of information that is useful is to know what it defaults to.

https://github.com/beatrichartz/parallel_stream/blob/main/lib/parallel_stream/defaults.ex#L3

it defaults to the number of schedulers you have * 2! and in our case schedulers is the number of cores on our machines. This is less than desirable for us because it gives unexpected parallelism on different nodes. But further, it's pretty sneaky that it's not even mentioned what it defaults to.
